jasmine/spec/inverted-index-test.js

```diff
@@ -1,53 +1,67 @@
-/* Setups goes here */
-const sampleBookData = [
+/* Sample Test Data */
+
+const book = [
   {
-    "title": "Alice in Wonderland",
-    "text": "Alice falls into a rabbit hole and enters a world full of imagination."
+    title: 'Alice in Wonderland',
+    text: 'Alice falls into a rabbit hole and enters a world full of ...'
   },
 
   {
-    "title": "The Lord of the Rings: The Fellowship of the Ring.",
-    "text": "An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring."
+    title: 'The Lord of the Rings: The Fellowship of the Ring.',
+    text: 'An unusual alliance of man, elf, dwarf, wizard and hobbit ...'
   }
 ];
 
-describe("Inverted Index Tests", () => {
+/* Test Setup */
+const myInvertedIndex = new InvertedIndex();
+myInvertedIndex.storeFile('book.json', book);
+myInvertedIndex.createIndex('book.json');
+const indexMap = myInvertedIndex.indexMap;
+const aliceSearchResult = { alice: { 'book.json': [0] } };
 
-  describe("Read Book Data", () => {
-    myInvertedIndex = new InvertedIndex();
-    myInvertedIndex.createIndex(sampleBookData);
+/* Test Suites */
+describe('Inverted Index Test Suites', () => {
+  describe('Read Book Data', () => {
+    it('should return false when checking a bad JSON array', () => {
+      expect(myInvertedIndex.isValidJSON(''))
+        .toBeFalsy();
+    });
 
-    it("should check that JSON is not empty", () => {
-      expect(myInvertedIndex.isValidJson([])).toBeFalsy();
+    it('should return true when validating a good JSON array', () => {
+      expect(myInvertedIndex.isValidJSON(book))
+        .toBeTruthy();
     });
 
-    it("should check that content of JSON is valid", () => {
-      expect(myInvertedIndex.isValidJson(sampleBookData)).toBeTruthy();
+    it('should return correct keys for files when file is saved', () => {
+      expect(Object.keys(myInvertedIndex.files))
+        .toEqual(['book.json']);
     });
 
+    it('should ensure the file content is saved accurrately', () => {
+      expect(myInvertedIndex.files['book.json'])
```

Comment:
You should consider setting no-undef to false in your .eslintrc to avoid this unnecessary hound comments